### PR TITLE
MIMXRT1024 FlexPWM Support

### PR DESCRIPTION
**Pull request by EmilioCBen**
Adding pinctrl to support the FlexPWM on the MIMXRT1024

### CODE REVIEW

File: tests/drivers/pwm/pwm_api/boards/mimxrt1024_evk.overlay [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/ {
+	aliases {
+		pwm-0 = &flexpwm1_pwm2;
+    };
+};
+
+&flexpwm1_pwm2 {
+	pinctrl-0 = <&pinmux_flexpwm1>;
+    pinctrl-names = "default";
```
Comment (jeppenodgaard) [inline tests/drivers/pwm/pwm_api/boards/mimxrt1024_evk.overlay:9]: Indent with tabs here

File: tests/drivers/pwm/pwm_api/boards/mimxrt1024_evk.overlay [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    status = "okay";
```
Comment (jeppenodgaard) [inline tests/drivers/pwm/pwm_api/boards/mimxrt1024_evk.overlay:10]: And tabs here

**Review by jeppenodgaard — CHANGES_REQUESTED** [review]

File: tests/drivers/pwm/pwm_api/boards/mimxrt1024_evk.overlay (hunk 1 above)
Comment (EmilioCBen) [reply tests/drivers/pwm/pwm_api/boards/mimxrt1024_evk.overlay]: Ahh yes, sorry about that...you know how text editors can be..should be fixed now.

File: tests/drivers/pwm/pwm_api/boards/mimxrt1024_evk.overlay (hunk 2 above)
Comment (EmilioCBen) [reply tests/drivers/pwm/pwm_api/boards/mimxrt1024_evk.overlay]: Also fixed!

**Review by decsny — DISMISSED** [review]

File: tests/drivers/pwm/pwm_api/boards/mimxrt1024_evk.overlay (hunk 1 above)
Comment (decsny) [reply tests/drivers/pwm/pwm_api/boards/mimxrt1024_evk.overlay]: you must use emacs

File: boards/arm/mimxrt1024_evk/mimxrt1024_evk.yaml [hunk 3]
```diff
@@ -24,3 +24,4 @@ supported:
   - sdhc
   - adc
   - usb_device
+  - flexpwm
```
Comment (danieldegrasse) [inline boards/arm/mimxrt1024_evk/mimxrt1024_evk.yaml:27]: this should be `- pwm` to trigger PWM testcases: https://github.com/zephyrproject-rtos/zephyr/blob/15c77acfb4326b7c39f3d2d48f13fbd9fe30404e/tests/drivers/pwm/pwm_api/testcase.yaml#L9

Comment (EmilioCBen) [reply boards/arm/mimxrt1024_evk/mimxrt1024_evk.yaml]: Done!

File: tests/drivers/dma/chan_link_transfer/boards/mimxrt1024_evk.overlay [hunk 4]
```diff
@@ -0,0 +1,7 @@
+/*
+ * Copyright (c) 2022 Kumar Gala <galak@kernel.org>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+dma0: &edma0 { };
```
[the remark this answers is not in the document]
Comment (EmilioCBen) [reply tests/drivers/dma/chan_link_transfer/boards/mimxrt1024_evk.overlay]: Sorry, fixed

**Review by danieldegrasse — APPROVED** [review]

**Review by jeppenodgaard — APPROVED** [review]

**Review by decsny — APPROVED** [review]

**Pull request merged**